RundeckAutomation/Functions/Start-RundeckJob.ps1

```diff
@@ -0,0 +1,41 @@
+function Start-RundeckJob
+{
+    <#
+    .SYNOPSIS
+    Starts a job from Rundeck.
+
+    .DESCRIPTION
+    The `Start-RundeckJob` function starts a job from Rundeck.
+
+    .EXAMPLE
+    Start-RundeckJob -ID 'b090d4c8-585c-4330-8bc6-ad4783089dfd'
+
+    Demonstrates how to start a specific job usng its ID.
+    #>
+    param(
+        [Parameter(Mandatory,ValueFromPipeline,ValueFromPipelineByPropertyName,ParameterSetName='ByID')]
+        [guid]
+        # The job's ID.
+        $ID,
+
+        [switch]
+        # Wait for job to complete or fail.
+        $Wait
+    )
+
+    process
+    {
+        $jobRun = Invoke-RundeckRestMethod -Method 'POST' -ResourcePath "job/$($ID)/executions"
+        Start-Sleep -Seconds 1

```

Comment:
What state are you waiting for? Can you call the API to find out if the job is in that state instead of sleeping?

Reply:
I think I was seeing a brief pause (sometimes) that has the execution queued but not accepted yet.  I'll do some more testing to see if this is actually needed or if there is a better way to handle this.